include/storage/flash_map.h

```diff
@@ -67,6 +68,12 @@ struct flash_area {
 	const char *fa_dev_name;
 };
 
+#define FLASH_AREA_TYPE_UNSPECIFIED			0x00 /* No type */
+#define FLASH_AREA_TYPE_MCUBOOT				0x01 /* zephyr,mcuboot,image */
```

Comment:
This does not look right. Should every bootloader place its ID here?
Disk subsystem offers possibility to find a block device without such hard-coded IDs.

Reply:
> 
> This does not look right. Should every bootloader place its ID here?

No, the ID should be added when support for bootloader within Zephyr is added.

> 
> Disk subsystem offers possibility to find a block device without such hard-coded IDs.

Yes, but the disk subsystem may use for example names of mount points.
I am trying to avoid bringing the "compatible" identifiers into flash; the space on mcuboot partition becomes tight.

Reply:
> Yes, but the disk subsystem may use for example names of mount points.
> I am trying to avoid bringing the "compatible" identifiers into flash; the space on mcuboot partition becomes tight.

How the disk subsystem identifiers are formed is debatable and not set in stone. This approach here does not scale and is not generic, like so much around flash.